components/admin/AdminShell.tsx

```diff
@@ -103,7 +116,15 @@ export default function AdminShell({
             placeholder='Select a competition'
             data={data?.map((c) => ({ value: c.code, label: c.name }))}
             value={comp}
-            onChange={(comp) => setComp(comp)}
+            onChange={(comp) => {
```

Comment:
My fault but avoid confounding variable names. Use `c` here (notice, `comp` is also a variable).